### ISSUE TITLE
Install issues on Mac OS X Sierra.

### ISSUE THREAD

**Issue by rememberkarl** [smartdevicelink/sdl_core#1302]
Hi all 'sdl_core' people!

I'm trying to install sdl_core, but I had some issues on it.
First issue was occurred when I tried 'cmake' command. 
'cmake' command was failed because of line number 211 of 'CMakeLists.txt', it was about 'Udev' library.
I just skipped this line and continued to install it, and I had narrowly done it.

**EDIT: I modified an above issue. I changed UDev library to libusb downloaded from Homebrew. And I made cmake file for libusb, so cmake routine worked well. But I'm not sure of that there will be no troubleshooting by this solution.**

But second issue was popped up when I tried 'make' command.
I don't even know what this problem is and requires what kind of solution or library.
Below is a log after 'make' command:

Scanning dependencies of target jsoncpp
[  1%] Building CXX object src/3rd_party-static/jsoncpp/CMakeFiles/jsoncpp.dir/src/lib_json/json_value.cpp.o
[  1%] Building CXX object src/3rd_party-static/jsoncpp/CMakeFiles/jsoncpp.dir/src/lib_json/json_writer.cpp.o
[  1%] Building CXX object src/3rd_party-static/jsoncpp/CMakeFiles/jsoncpp.dir/src/lib_json/json_reader.cpp.o
[  1%] Linking CXX static library libjsoncpp.a
[  1%] Built target jsoncpp
Scanning dependencies of target 3rd_party_logger
 Need to build logger. 
Scanning dependencies of target expat
[  0%] Building C object src/3rd_party/expat-2.1.0/CMakeFiles/expat.dir/lib/xmlparse.c.o
[  0%] Building C object src/3rd_party/expat-2.1.0/CMakeFiles/expat.dir/lib/xmlrole.c.o
[  0%] Building C object src/3rd_party/expat-2.1.0/CMakeFiles/expat.dir/lib/xmltok.c.o
[  0%] Building C object src/3rd_party/expat-2.1.0/CMakeFiles/expat.dir/lib/xmltok_impl.c.o
[  0%] Building C object src/3rd_party/expat-2.1.0/CMakeFiles/expat.dir/lib/xmltok_ns.c.o
[ 50%] Linking C shared library libexpat.dylib
[ 50%] Built target expat
Scanning dependencies of target libapr-1
[ 50%] Generating ../apr-build/Makefile
checking build system type... x86_64-apple-darwin16.4.0
checking host system type... x86_64-apple-darwin16.4.0
checking target system type... x86_64-apple-darwin16.4.0
Configuring APR library
Platform: x86_64-apple-darwin16.4.0
checking for working mkdir -p... yes
APR Version: 1.5.0
checking for chosen layout... apr
checking for gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/Library/SDL/trunk/build/src/3rd_party/apr-build':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
make[5]: *** [src/3rd_party/apr-build/Makefile] Error 1
make[4]: *** [src/3rd_party/apr-cmake/CMakeFiles/libapr-1.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [CMakeFiles/3rd_party_logger] Error 2
make[1]: *** [CMakeFiles/3rd_party_logger.dir/all] Error 2
make: *** [all] Error 2

Please give me the answers or precise solutions..

Best regards.
Karl

**Comment by jacobkeeler** [comment]
Hello @rememberkarl, the reason that you are getting these errors is because the SDL Core project hasn't been tested on Mac OS X, as the project does not officially support building on this system. As it stands, the only supported system for building SDL Core is Ubuntu (though there are plans for a Windows port).

To my knowledge there are no current plans to create a port for OS X in the future, so if you really need to get Core running on your system, you will need to either:
1. Create an Ubuntu Virtual Machine and run SDL Core through that setup, or
2. Write your own OS X port for SDL Core (in which case, it would be appreciated if you commit your port back to this repository upon its completion)

**Comment by rememberkarl** [comment]
Hi @jacobkeeler,

Thank you for your comment.
Currently I'm trying to figure out from the beginning which part of this library occurs some issues on Mac OS X, so I'll give you some ideas, hopefully, and after all integration and test on Mac OS X I'll commit my own works for a contribution. (But so many files and codes, you know... ^^;)

Best regards,
Karl

**Comment by davidswi** [comment]
Hi all, I have an experimental port of SDL Core to OSX. I am planning to clean it up and submit it as a PR.

**Comment by davidswi** [comment]
@rememberkarl I took the approach of exporting the CMake scripts to an XCode workspace as the first step. IMHO, the principal advantage of running SDL Core on OSX is for the rich code analysis and debugging that XCode provides. I also disabled all transports except TCP/IP. I want to clean this up and abstract the BT and USB transport implementations using a PAL that will allow SDL Core to be easily ported to and maintained on OSX, Windows and other platforms that deviate from the Linux device models for USB and BT.
